app.territoiresentransitions.react/src/ui/shared/preuves/AddPreuveModal/AddFromLib.tsx

```diff
@@ -24,103 +16,67 @@ export type TAddFromLibProps = {
  * et permet de sélectionner les fichiers à ajouter comme preuve
  */
 export const AddFromLib = (props: TAddFromLibProps) => {
-  const {
-    items: fichiers,
-    total,
-    filters,
-    onAddFileFromLib,
-    onClose,
-    setFilters,
-  } = props;
+  const {items: fichiers, onAddFileFromLib, onClose, setFilters} = props;
 
-  const [currentSelection, setCurrentSelection] = useState<{
-    [key: number]: boolean;
-  }>({});
-
-  const tracker = useFonctionTracker();
+  const [selectedFiles, setSelectedFiles] = useState<Option[] | undefined>();
+  const [options, setOptions] = useState<Option[]>(
+    fichiers.map(f => ({label: f.filename, value: f.id}))
+  );
 
-  const canAdd = Object.values(currentSelection).find(v => !!v);
+  useEffect(() => {
+    const remainingOptions = fichiers
+      .filter(f => !(selectedFiles ?? []).some(file => file.value === f.id))
+      .map(f => ({label: f.filename, value: f.id}));
 
-  // met à jour notre sélection interne lorsqu'une case est dé/cochée
-  const onChange = (e: ChangeEvent<HTMLFieldSetElement>) => {
-    if (e.target.tagName === 'INPUT' && e.target.type === 'checkbox') {
-      const {target} = e as unknown as ChangeEvent<HTMLInputElement>;
-      const {checked, id} = target;
-      setCurrentSelection({...currentSelection, [id]: checked});
-    }
-  };
+    setOptions([...(selectedFiles ?? []), ...remainingOptions]);
+  }, [JSON.stringify(fichiers), JSON.stringify(selectedFiles)]);
```

Comment:
J'arrive pas bien à comprendre à quoi sert ce `useEffect` ici ?

Reply:
C'est utile dans le cas d'une recherche et / ou sélection d'un fichier qui n'est pas dans la liste reçue en prop (soit les cinq premiers éléments qui matchent avec le filtre)

Reply:
Je pense qu'on peut réaliser la même chose sans `useState` et sans `useEffect`. Ici on est typiquement dans le cas de transformation de données qui peuvent être faite sans `useEffect`, ce qui est préférable. 

[Cette page de la doc React](https://react.dev/learn/you-might-not-need-an-effect) est toujours un bon rappel que bien des fois on peut faire les choses plus simplement que nos vieux réflexes React. Je retombe dessus régulièrement quand j'essaye de simplifier du code React 🙂


À valider niveau comportement, j'ai fait quelques essais qui me semble concluants, mais on parvient à faire la même chose avec ça :

```
// On garde ce `useState` nécessaire
  const [selectedOptions, setSelectedOptions] = useState<Option[]>([]);

  // on mappe les fichiers reçus en props au format Option
  const availableOptions = fichiers.map(f => ({
    label: f.filename,
    value: f.id,
  }));

  // on combine les options sélectionner et disponible
  // en dédupliquant pour pas avoir de doublon quand l'option existe dans les deux arrays
  const options = uniqBy([...availableOptions, ...selectedOptions], f => f.value);

```


Avec une petite fonction `uniqBy` à importer d'une lib extérieure, ou bien qu'on peut mettre juste ici dans le fichier : 

```
function uniqBy<T, U>(arr: T[], mapper: (item: T) => U): T[] {
  return arr.filter((item, index, self) => {
    return index === self.findIndex(t => mapper(t) === mapper(item));
  });
}
```

Je te laisse me dire ce que tu en penses, si tu veux que j'ajoute ces modifs dans un commit tu me dis.

Reply:
J'ai mis cette partie à jour du coup